databuilder/extractor/dashboard/mode_analytics/mode_dashboard_last_successful_executions_extractor.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from pyhocon import ConfigTree, ConfigFactory  # noqa: F401
+
+from databuilder.extractor.dashboard.mode_analytics.mode_dashboard_executions_extractor import \
+    ModeDashboardExecutionsExtractor
+from databuilder.extractor.dashboard.mode_analytics.mode_dashboard_utils import ModeDashboardUtils
+from databuilder.extractor.restapi.rest_api_extractor import STATIC_RECORD_DICT
+from databuilder.models.dashboard.dashboard_execution import DashboardExecution
+from databuilder.rest_api.rest_api_query import RestApiQuery
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ModeDashboardLastSuccessfulExecutionExtractor(ModeDashboardExecutionsExtractor):
+    """
+    A Extractor that extracts Mode dashboard's last successful run (execution) timestamp.
+
+    """
+
+    def __init__(self):
+        super(ModeDashboardLastSuccessfulExecutionExtractor, self).__init__()
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+
+        conf = conf.with_fallback(
+            ConfigFactory.from_dict({
+                STATIC_RECORD_DICT: {'product': 'mode',
+                                     'execution_state': 'succeeded',
```

Comment:
should be put the execution_state for mode as a enum which could be used in multiple places?

Reply:
Actually, it was intentional not to go with enum as we don't know what state other vendor (Tableau, Superset, etc) would provide

Reply:
I am referring putting an enum just within mode_analytics folder which could share across other mode executors.

Reply:
Yeah, I usually like the idea of using enum but there're couple things that made me leaning toward to just string. First one is that Mode does not officially publish the enum of state on their doc so I am not sure we can validate it. (https://mode.com/developer/api-reference/analytics/reports/). The other one is that I think it's anyway better to represent the state of source unless we want to standardize state among Dashboard.  WDYT? 😄

Reply:
Yeah, I usually like the idea of using enum but there're couple things that made me leaning toward to just string. First one is that Mode does not officially publish the enum of state on their doc so I am not sure we can validate it. (https://mode.com/developer/api-reference/analytics/reports/). The other one is that I think it's anyway better to represent the state of source unless we want to standardize state among Dashboard. WDYT? 😄